ark/phenotyping/som_utils.py

```diff
@@ -131,50 +111,152 @@ def compute_pixel_cluster_channel_avg(fovs, channels, base_dir, cluster_col,
-                                     fovs=None, cluster_col='cluster'):
-    """Compute the average marker expression for each cell
+def compute_cell_cluster_channel_avg(fovs, channels, base_dir,
+                                     weighted_cell_channel_avg_name,
+                                     cell_cluster_name='cell_mat_clustered.feather',
+                                     cell_cluster_col='cell_meta_cluster'):
+    """Computes the average marker expression for each cell cluster
+
+    Args:
+        fovs (list):
+            The list of fovs to subset on
+        channels (list):
+            The list of channels to subset on
+        base_dir (str):
+            The path to the data directory
+        weighted_cell_channel_avg_name (str):
+            The name of the weighted cell table, created in `example_cell_clustering.ipynb`
+        cell_cluster_name (str):
+            Name of the file containing the cell data with cluster labels
+        cell_cluster_col (str):
+            Whether to aggregate by cell SOM or meta labels
+            Needs to be either 'cell_som_cluster', or 'cell_meta_cluster'
+
+    Returns:
+        pandas.DataFrame:
+            Each cell cluster mapped to the average expression for each marker
+    """
+
+    # verify the cell table actually exists
+    if not os.path.exists(os.path.join(base_dir, weighted_cell_channel_avg_name)):
+        raise FileNotFoundError(
+            "Weighted cell table %s not found in %s" % (weighted_cell_channel_avg_name, base_dir)
+        )
+
+    # verify the cell cluster col specified is valid
+    misc_utils.verify_in_list(
+        provided_cluster_col=[cell_cluster_col],
+        valid_cluster_cols=['cell_som_cluster', 'cell_meta_cluster']
+    )
+
+    cell_table = pd.read_csv(os.path.join(base_dir, weighted_cell_channel_avg_name))
+
+    # subset on only the fovs the user has specified
+    cell_table = cell_table[cell_table['fov'].isin(fovs)]
+
+    # because current version of pandas doesn't support key-based sorting, need to do it this way
+    cell_table['fov'] = cell_table['fov'].map(lambda x: x.replace('fov', '')).astype(int)
+    cell_tabel = cell_table.sort_values(by=['fov', 'segmentation_label']).reset_index(drop=True)
+    cell_table['fov'] = cell_table['fov'].map(lambda x: 'fov' + str(x))
+
+    # read the clustered data
+    cluster_data = feather.read_dataframe(os.path.join(base_dir, cell_cluster_name))
+
+    # assign the cluster labels to cell_table
+    cell_table[cell_cluster_col] = cluster_data[cell_cluster_col]
+
+    # subset the cell table by just the desired channels and the cell_cluster_col
+    cell_table = cell_table[channels + [cell_cluster_col]]
+
+    # compute the mean channel expression across each cell cluster
+    cluster_avgs = cell_table.groupby(cell_cluster_col).mean().reset_index()
+
+    # z-score the cluster_avgs
+    cluster_avgs[channels] = stats.zscore(cluster_avgs[channels].values)
```

Comment:
Not sure if I missed it, but I don't think you have an analogous z-scoring line for the pixel clusters. Probably makes sense to add it to compute_pixel_custer_channel_avg.